node/bin/src/lib.rs

```diff
@@ -705,9 +709,19 @@ async fn run_en_pipeline<
             repositories: repositories.clone(),
             sequencer_config: config.sequencer_config.clone().into(),
         })
-        .pipe(TreeManager { tree: tree.clone() })
-        .pipe(NoOpSink::new())
-        .spawn(tasks);
+        .pipe(TreeManager { tree: tree.clone() });
+    let pipeline = if config.batch_verification_config.enabled {
+        pipeline.pipe(BatchVerificationClient::new(
+            config.batch_verification_config.signing_key.clone(),
+            config.genesis_config.chain_id.unwrap(), // TODO check unwrap
+            node_state_on_startup.l1_state.diamond_proxy,
+            config.batch_verification_config.address,
+        ))
+    } else {
+        pipeline.pipe(NoOpSink::new())
+    };
```

Comment:
This can be simplified if you pipe `Option<BatchVerificationClient>` before `NoOpSink`. See my comment [here](https://github.com/matter-labs/zksync-os-server/pull/525#discussion_r2438021671)

Reply:
Hi, looked into this. So I would need to add that implementation for Option<C>, but to have a sensible implementation for NAME and OUTPUT_BUFFER_SIZE I would need to change those from const to functions. Can we push this to a separate PR, becouse its going to touch a lot of files?

Reply:
yes, it can be improved but it's ok to do separately